javascript/ui/page/passwordsignup.js

```diff
@@ -43,6 +44,7 @@ goog.require('firebaseui.auth.ui.page.Base');
  */
 firebaseui.auth.ui.page.PasswordSignUp = function(
```

Comment:
Apparently I can't comment lines too far away from changes, so I'm leaving my comment here...

You should also change the enterDocument and setupFocus_ method logic below to not initialize the name component or attach event listener on it when shouldDisplayName is false.

This will also avoid having the logic to ensure the name element is rendered when calling initNameElement (ref: a previous comment).